hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -415,12 +422,16 @@ public void initialize(URI name, Configuration originalConf)
       // DT Bindings may override this
       setEncryptionSecrets(new EncryptionSecrets(
           getEncryptionAlgorithm(bucket, conf),
-          getServerSideEncryptionKey(bucket, getConf())));
+          getS3EncryptionKey(bucket, getConf())));
 
       invoker = new Invoker(new S3ARetryPolicy(getConf()), onRetry);
       instrumentation = new S3AInstrumentation(uri);
       initializeStatisticsBinding();
-
+      // If CSE-KMS method is set then CSE is enabled.
+      isCSEEnabled = S3AUtils.lookupPassword(conf,
```

Comment:
@mehakmeet : I was just reviewing this PR to get some understanding. 
Wondering why the `isCSEEnabled` check only compares if the `SERVER_SIDE_ENCRYPTION_ALGORITHM` param is not equal to null. Shouldn't it be: (identical to the one in `DefaultS3ClientFactory.java`)
```
if (S3AEncryptionMethods.getMethod(S3AUtils.
          lookupPassword(conf, SERVER_SIDE_ENCRYPTION_ALGORITHM, null))
          .equals(S3AEncryptionMethods.CSE_KMS)) {
```

I see that this check is changed in `trunk`(to compare equal to `CSE-KMS` but making sure I am not missing something here

Reply:
Hey @balvisio. Yes, this was incorrect and fixed in subsequent patches related to S3A CSE. In the current branch it's:
```
isCSEEnabled = S3AEncryptionMethods.CSE_KMS.getMethod()
          .equals(getS3EncryptionAlgorithm().getMethod());
```